### PR TITLE
Documentation Updates

### PR DESCRIPTION
**Pull request by cmsirbu**
## What's changed

- Moved existing documentation into a new structure to prepare for the mkdocs rendered docsite.
- Edited and improved the visuals in various places using new syntax from the [Material for Mkdocs theme](https://squidfunk.github.io/mkdocs-material/reference/).
- Added configuration for building and hosting the docsite on RTD.
- Replaced the project README with a standardized one. Note: the README will have certain links/images broken until the PR is merged into `develop`.

## TODO

Some sections throughout the files listed below require input (or removal if not applicable to this plugin). All `Developer Note` comments should be resolved and removed.

- [ ] user/app_overview.md
- [ ] user/app_use_cases.md
- [ ] user/app_getting_started.md
- [ ] user/faq.md
- [ ] user/external_interactions.md
- [ ] admin/install.md - review (most old README content) ported here
- [ ] admin/upgrade.md
- [ ] admin/uninstall.md
- [ ] admin/compatiblity_matrix.md
- [ ] admin/release_notes - I think the CHANGELOG content should be linking to the docs section release notes, but will leave this decision up to you.
- [ ] dev/extending.md - review
- [ ] dev/contributing.md
- [ ] dev/dev_environment.md - boilerplate guide based on cookie template (old GETTING_STARTED guide). A sync to the cookie dev environment would be beneficial to ensure instructions are accurate.
- [ ] dev/code_reference - check examples provided and add any module/package auto-generated docs that are relevant
- [ ] dev/arch_decision.md
- [ ] README.md - review its contents!

Other issues:

- [ ] Added test to check that RTD used `docs/requirements.txt` are in sync with dependencies in `pyproject.toml` - needs the following item fixed first, might need to remove it?
- [ ] Couldn't add docs dependencies (see `docs/requirements.txt` for exact versions) to poetry environment for development because of Jinja2 conflict with Nautobot 1.2.

```
And because nautobot (1.2.11) depends on Jinja2 (>=2.11.3,<2.12.0)
   and mkdocs-material (8.4.2) depends on jinja2 (>=3.0.2), mkdocs-material (8.4.2) is incompatible with nautobot (>=1.2.1,<1.3.0).
```

### CODE REVIEW

**Comment by bryanculver** [comment]
Dependent on #56 as it will update the requirements in this repo.

File: docs/admin/uninstall.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
+# Uninstall the App from Nautobot
+
+Here you will find any steps necessary to cleanly remove the App from your Nautobot environment.
+
+## Uninstall Guide
```
Comment (bryanculver) [inline docs/admin/uninstall.md:5]: I think we should do a "write once" guide in Core for this.

Any one-offs can be noted here.

File: docs/admin/upgrade.md [hunk 2]
```diff
@@ -0,0 +1,10 @@
+# Upgrading the App
+
+Here you will find any steps necessary to upgrade the App in your Nautobot environment.
+
+## Upgrade Guide
```
Comment (bryanculver) [inline docs/admin/upgrade.md:5]: As with the Uninstall case: I think we should do a "write once" guide in Core for this.

Any one-offs can be noted here.

File: docs/dev/contributing.md [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+## Branching Policy
+
+!!! warning "Developer Note - Remove Me!"
+    What branching policy is used for this project and where contributions should be made.
+
+## Release Policy
+
+!!! warning "Developer Note - Remove Me!"
+    How new versions are released.
```
Comment (bryanculver) [inline docs/dev/contributing.md:24]: I think we should do a "write once" guide in Core for this.

Any one-offs can be noted here.

**Comment by DistantVoyager** [comment]
These changes are handled in #134

**Pull request closed without merging**